src/IODebug.jl

```diff
@@ -25,87 +25,92 @@ else
 
 end
 
-Base.wait_close(iod::IODebug) = Base.wait_close(iod.io)
-
-Base.write(iod::IODebug, a...) =
-    (logwrite(iod, :write, join(a));
-     write(iod.io, a...))
-
-Base.write(iod::IODebug, a::Array) =
-    (logwrite(iod, :write, join(a));
-     write(iod.io, a))
-
-Base.write(iod::IODebug, x::String) =
-    (logwrite(iod, :write, x);
-     write(iod.io, x))
-
-Base.unsafe_write(iod::IODebug, x::Ptr{UInt8}, n::UInt) =
-    (logwrite(iod, :unsafe_write, unsafe_string(x,n));
-     unsafe_write(iod.io, x, n))
-
-Base.unsafe_read(iod::IODebug, x::Ptr{UInt8}, n::UInt) =
-    (r = unsafe_read(iod.io, x, n);
-     logread(iod, :unsafe_read, unsafe_string(x,n)); r)
-
-Base.read(iod::IODebug, n::Integer) =
-    (r = read(iod.io, n);
-     logread(iod, :read, String(copy(r))); r)
-
-Base.readavailable(iod::IODebug) =
-    (r = readavailable(iod.io);
-     logread(iod, :readavailable, String(copy(r))); r)
-
-Base.readuntil(iod::IODebug, f) =
-    (r = readuntil(iod.io, f);
-     logread(iod, :readuntil, String(copy(r))); r)
-
-Base.readuntil(iod::IODebug, f, h) =
-    (r = readuntil(iod.io, f, h);
-     logread(iod, :readuntil, String(copy(r))); r)
-
-Base.eof(iod::IODebug) = eof(iod.io)
-Base.close(iod::IODebug) = close(iod.io)
-Base.isopen(iod::IODebug) = isopen(iod.io)
-Base.iswritable(iod::IODebug) = iswritable(iod.io)
-Base.isreadable(iod::IODebug) = isreadable(iod.io)
-IOExtras.startread(iod::IODebug) = startread(iod.io)
-IOExtras.startwrite(iod::IODebug) = startwrite(iod.io)
-IOExtras.closeread(iod::IODebug) = closeread(iod.io)
-IOExtras.closewrite(iod::IODebug) = closewrite(iod.io)
-
-Base.bytesavailable(iod::IODebug) = bytesavailable(iod.io)
-
-Base.show(io::IO, iod::IODebug) = show(io, iod.io)
-
-function show_log(io::IO, iod::IODebug)
-    lock(io)
-    println(io, "$(typeof(iod)):\nio:     $(iod.io)")
-    prevop = ""
-    for (operation, f, bytes) in iod.log
-        if prevop != "" && prevop != operation
-            println(io)
+debug_mode()::Bool = get(ENV, "HTTP_DEBUG", "") == "true"
```

Comment:
Why introduce yet another "debug lever" in an ENV variable vs. just defining these if `DEBUG_LEVEL[] > 0`? I know it's a bit of a backward step since we changed `DEBUG_LEVEL` from a const to a Ref so it could be dynamic, but whatevs, it's just a debug mechanism. I'd rather use an existing lever vs. introduce a new one

Reply:
I've just now tried that. It does work but I'm unable to allow downstream packages to enable debugging. They have to override the constant before the module is loaded.

Reply:
The common trick to set the enviroment variable in the `__init__()` of the downstream package doesn't work either.

Reply:
A more future-proof solution could be to define intermediate methods such as `httpwrite` where
```julia
http_write(io::IODebug) = ...
http_write(io::IO) = Base.write(...)
```
This way, `IODebug` doesn't need to be a subtype of `IO` and therefore the method invalidations in `Base` will also be avoided.